### PR TITLE
Add `backgroundReload` option to `store.findRecord` & `store.findAll`

### PR DESCRIPTION
**Pull request by HeroicEric**
Fixes https://github.com/emberjs/data/issues/3772

This adds support for controlling when records are reloaded in the
background to `store.findRecord()` and `store.findAll()`.

### store.findRecord()

If you would like to force or prevent background reloading, you can set a
boolean value for `backgroundReload` in the options object for
`findRecord`.

```javascript
import Ember from 'ember';

export default Ember.Route.extend({
  model: function(params) {
    return this.store.findRecord('post', params.post_id, { backgroundReload: false });
  }
});
```

### store.findAll()

If you would like to force or prevent background reloading, you can set a
boolean value for `backgroundReload` in the options object for
`findAll`.

```javascript
import Ember from 'ember';

export default Ember.Route.extend({
  model: function() {
    return this.store.findAll('post', { backgroundReload: false });
  }
});
```

### CODE REVIEW

**Comment by HeroicEric** [comment]
@pangratz ask and ye shall receive

**Comment by pangratz** [comment]
😍  awesome @HeroicEric, I will review on monday!

File: addon/-private/system/store.js [hunk 1]
```diff
@@ -547,16 +565,21 @@ Store = Service.extend({
       return this.scheduleFetch(internalModel, options);
     }
 
-    // Refetch the record if the adapter thinks the record is stale
+    if (options.backgroundReload === false) {
+      return Promise.resolve(internalModel);
+    }
```
Comment (pangratz) [inline addon/-private/system/store.js]: I tend to think that this needs to be moved below the code which checks if a reload should be done (either via the options or via the adapters' `shouldReloadRecord`). In my understanding a reload should take precedence over a background reload.

IMHO the logic should be:

- if we should do a reload (either configured via the passed option or via the adapters' `shouldReloadRecord`), then return a promise which resolves once the record has been reloaded
- if we shouldn't do a reload but we should do a background reload (either configured via the passed option or via , we return a promise which resolves immediately but the record is reloaded in the background.

What do you think?

Comment (HeroicEric) [inline addon/-private/system/store.js]: Agreed.

**Comment by pangratz** [comment]
Thanks @HeroicEric 🚀

**Pull request merged**